Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift

```diff
@@ -44,40 +44,45 @@ class ViewController: UIViewController {
     }
     
     @IBAction func acquireToken(_ sender:UIButton) {
-        let authContext = ADAuthenticationContext(authority: "https://login.microsoftonline.com/common",
-                                                  error: nil)
         
-        authContext!.acquireToken(withResource: "https://graph.windows.net",
-                                             clientId: "b92e0ba5-f86e-4411-8e18-6b5f928d968a",
-                                             redirectUri: URL(string: "urn:ietf:wg:oauth:2.0:oob"))
+        guard let authContext = ADAuthenticationContext(authority: "https://login.microsoftonline.com/common", error: nil) else {
+            
+            print("Failed to create auth context")
+            return
+        }
+        
+        authContext.acquireToken(withResource: "https://graph.windows.net",
+                                 clientId: "b92e0ba5-f86e-4411-8e18-6b5f928d968a",
+                                 redirectUri: URL(string: "urn:ietf:wg:oauth:2.0:oob")!)
```

Comment:
self needs to be referenced weak here.
option 1) 
guard self = self else { return } and then use self,
or self? for all the references.

Reply:
Addressed